static/js/compose_actions.js

```diff
@@ -243,6 +243,39 @@ exports.cancel = function () {
     $(document).trigger($.Event('compose_canceled.zulip'));
 };
 
+exports.respond_to_empty = function (opts) {
+    if (current_msg_list.selected_message()) {
+        // If message exists, use respond_to_message instead.
+        exports.respond_to_message(opts);
+        return;
+    }
+
+    var msg_type = 'stream'; // Set msg_type to stream by default
+                             // in the case of an empty home view.
+    var stream = '';
+    var subject = '';
+    var pm_recipient = '';
+
+    var operators = narrow.filter()._operators;
```

Comment:
You shouldn't be peeking at `_.operators()` here.  There should be more specific functions in `narrow.js` to find out if you are in a PM narrow, or a stream narrow, etc.  Also, I don't think we want to reply if the narrow is to things like `is:starred` or keyword search.